### PR TITLE
scripts/libraries: Update normalization to handle 3D/4D image tensors.

### PR DESCRIPTION
**Pull request by kwagyeman**
Tested with FOMO and YOLO.

### CODE REVIEW

**Comment by kwagyeman** [comment]
TODO: PR should be updated to just check if the channels are 1/3 and batch size is 1.

**Comment by iabdalkader** [comment]
This might be stay as it is if the batch size removal is not merged.

File: scripts/libraries/ml/ml/preprocessing.py [hunk 1]
```diff
@@ -35,8 +35,18 @@ def __call__(self, *args):
...
-        pixfmt = image.GRAYSCALE if shape[3] == 1 else image.RGB565
-        img = image.Image(shape[2], shape[1], pixfmt, buffer=buffer)
+        if len(shape) == 3:
+            h, w, c = shape
+        elif len(shape) == 4:
+            b, h, w, c = shape
+            if b != 1:
+                raise ValueError("Expected 1 image per batch")
+        else:
+            raise ValueError("Expected 3D or 4D input tensor ([batch,] height, width, channels)")
```
Comment (iabdalkader) [inline scripts/libraries/ml/ml/preprocessing.py:45]:
```suggestion
        if len(shape) != 4 or shape[0] != 1:
                raise ValueError("Expected ([batch,] height, width, channels)")
        b, h, w, c = shape
```
This for sure will always be `b, h, w, c`

**Review by iabdalkader — CHANGES_REQUESTED** [review]

File: scripts/libraries/ml/ml/preprocessing.py (hunk 1 above)
Comment (kwagyeman) [reply scripts/libraries/ml/ml/preprocessing.py:45]: Yes, will reduce it to just sanity checks.

File: scripts/libraries/ml/ml/preprocessing.py [hunk 2]
```diff
@@ -35,8 +35,15 @@ def __call__(self, *args):
 
         buffer, shape, dtype = args
         # Create an image using the input tensor as buffer.
-        pixfmt = image.GRAYSCALE if shape[3] == 1 else image.RGB565
-        img = image.Image(shape[2], shape[1], pixfmt, buffer=buffer)
+        if len(shape) != 4:
+            raise ValueError("Expected 4D input tensor (batches, height, width, channels)")
```
Comment (iabdalkader) [inline scripts/libraries/ml/ml/preprocessing.py:39]:
```suggestion
            raise ValueError("Expected input tensor with shape: (1, H, W, C)")
```

**Review by iabdalkader — APPROVED** [review]

**Pull request merged**